build/scripts/deploy-msbuild.ps1

```diff
@@ -0,0 +1,51 @@
+
+# This script will ensure that the MSBuild toolset of the specified 
+# version is downloaded and deployed to Binaries\Toolset\MSBuild
+#
+# This is a no-op if it is already present at the appropriate version
+
+param ()
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+$version = "0.1.7-alpha"
+$repoDir = [IO.Path]::GetFullPath((join-path $PSScriptroot "..\.."))
+$toolsDir = join-path $repoDir "Binaries\Toolset"
+$scratchDir = join-path $toolsDir "Scratch"
+
+try {
+
+    mkdir $toolsDir -ErrorAction silent | out-null
+    mkdir $scratchDir -ErrorAction silent | out-null
+    pushd $toolsDir
+
+    $scratchFile = join-path $scratchDir "msbuild.txt"
```

Comment:
@tmat 

It made more sense to me as well and hence I started with that.  But then you quickly find out that our unzip APIs get really upset about files that exist in the directory.  😦